### PR TITLE
feat: add AnyNode type

### LINKED ISSUE #9049: Add AnyNode type
### Describe the feature

similar to #9028 but for the `Node` trait 

https://github.com/paradigmxyz/reth/blob/42bbafff2c825a7c2a300e5c1156d08fed083162/crates/node/builder/src/node.rs#L23-L23

this can be solved similarly but with an actual instance of `C: NodeComponentsBuilder`

struct AnyNode<N =, C = ()>(PhantomType<N>, C);

### Additional context

FYI @leruaa

### PR DESCRIPTION
**Pull request by romanroibu**
Closes #9049

### CODE REVIEW

File: crates/node/builder/src/node.rs [hunk 1]
```diff
@@ -28,6 +28,32 @@ pub trait Node<N: FullNodeTypes>: NodeTypes + Clone {
     fn components_builder(self) -> Self::ComponentsBuilder;
 }
 
+/// A [`Node`] type builder
+#[derive(Clone, Default, Debug)]
+pub struct AnyNode<N = (), C = ()>(PhantomData<N>, C);
+
```
Comment (mattsse) [inline crates/node/builder/src/node.rs:34]: we also need two functions

1. set N
2. set C

**Review by mattsse — CHANGES_REQUESTED** [review]
nice, almost there

**Comment by romanroibu** [comment]
hey @mattsse, thanks for your review! please LMK if my latest change is what you had in mind. not completely sure about the naming, so if you have better suggestions - happy to update.

File: crates/node/builder/src/node.rs [hunk 2]
```diff
@@ -32,6 +32,18 @@ pub trait Node<N: FullNodeTypes>: NodeTypes + Clone {
 #[derive(Clone, Default, Debug)]
 pub struct AnyNode<N = (), C = ()>(PhantomData<N>, C);
 
+impl<N, C> AnyNode<N, C> {
+    /// Sets the `N: FullNodeTypes` generic type.
+    pub fn full_node_types<T>(self) -> AnyNode<T, C> {
```
Comment (mattsse) [inline crates/node/builder/src/node.rs:37]:
```suggestion
    pub fn types<T>(self) -> AnyNode<T, C> {
```

File: crates/node/builder/src/node.rs [hunk 3]
```diff
@@ -32,6 +32,18 @@ pub trait Node<N: FullNodeTypes>: NodeTypes + Clone {
 #[derive(Clone, Default, Debug)]
 pub struct AnyNode<N = (), C = ()>(PhantomData<N>, C);
 
+impl<N, C> AnyNode<N, C> {
+    /// Sets the `N: FullNodeTypes` generic type.
```
Comment (mattsse) [inline crates/node/builder/src/node.rs:36]:
```suggestion
    /// Configures the types of the node.
```

**Review by mattsse — APPROVED** [review]
lgtm, ty

**Pull request merged**